src/main/java/com/comcast/zucchini/AbstractZucchiniTest.java

```diff
@@ -132,6 +147,8 @@ private boolean envSerialized() {
      * @param contexts The list of contexts that will be run.
      */
     public void runParallel(List<TestContext> contexts) {
+        this.canBarrierInTest = this.canBarrier;
```

Comment:
With the API, runParallel() and runSerial() are both public methods.  Since they are public, the run() method is not always called.  In the case that someone decided to call the bare runParallel() and runSerial() methods, this method prevents bad behavior on the part of zucchini.  We _could_ put something in the docs about undefined behavior on using a barrier and calling the bare runParallel() and runSerial() methods, and then we could put the statement just in the run() method, but I think that's less stable.  

tl;dr - The current method of doing this is the belt-and-suspenders approach.  For the next api change, it might desirable to make the runParallel() and runSerial() private, and provide an overloaded run() method so that it defaults in accordance with envSerialized(), and can be overridden with run(boolean).